pkg/controllers/manila/manila.go

```diff
@@ -123,6 +136,62 @@ func (c *ManilaController) sync(ctx context.Context, syncCtx factory.SyncContext
 	return c.setEnabledCondition(ctx)
 }
 
+func (c *ManilaController) getFsGroupPolicy(ctx context.Context) storagev1.FSGroupPolicy {
+
+	// Set the fsGroupPolicy param to the default value
+	fsGroupPolicy := "None"
+
+	// Try to get the current value defined for the operator
+	if c.isFSGroupPolicyAllowed(os.Getenv("CSI_FSGROUP_POLICY")) {
```

Comment:
The logic is a bit confusing. IMO having a new function to validate the value makes it unnecessary complicate, we can do the validation here if we want to. We should also log for unknown values.

How about something like the following?
```go 
func (c *ManilaController) getFsGroupPolicy(ctx context.Context) storagev1.FSGroupPolicy {
	// NOTE: shoudn't the default be storagev1.ReadWriteOnceWithFSTypeFSGroupPolicy?
	fsGroupPolicy := storagev1.NoneFSGroupPolicy

	fsGroupPolicyFromEnv := storagev1.FSGroupPolicy(os.Getenv("CSI_FSGROUP_POLICY"))
	switch fsGroupPolicyFromEnv {
	case storagev1.NoneFSGroupPolicy, storagev1.FileFSGroupPolicy, storagev1.ReadWriteOnceWithFSTypeFSGroupPolicy:
		fsGroupPolicy = fsGroupPolicyFromEnv
	default:
		if fsGroupPolicyFromEnv != "" {
			klog.V(4).Infof("Invalid CSI_FSGROUP_POLICY %q. Ignoring.", fsGroupPolicyFromEnv)
		}
	}

	return fsGroupPolicy
}
```

Reply:
+1 thanks, the logic you're proposing here looks good to me. Updating the patch in a bit with a more simplified logic.